searchcore/src/tests/proton/reference/gid_to_lid_change_handler/gid_to_lid_change_handler_test.cpp

```diff
@@ -168,92 +178,92 @@ TEST_F("Test that we can register multiple listeners", Fixture)
     f.addListener(std::move(listener2));
     f.addListener(std::move(listener3));
-    TEST_DO(stats1.assertCounts(1, 1, 0, 0));
-    TEST_DO(stats2.assertCounts(1, 1, 0, 0));
-    TEST_DO(stats3.assertCounts(1, 1, 0, 0));
+    TEST_DO(stats1.assertListeners(1, 1, 0));
+    TEST_DO(stats2.assertListeners(1, 1, 0));
+    TEST_DO(stats3.assertListeners(1, 1, 0));
     f.notifyPut(toGid(doc1), 10, 10);
-    TEST_DO(stats1.assertCounts(1, 1, 0, 1));
-    TEST_DO(stats2.assertCounts(1, 1, 0, 1));
-    TEST_DO(stats3.assertCounts(1, 1, 0, 1));
+    TEST_DO(stats1.assertChanges(1, 0));
+    TEST_DO(stats2.assertChanges(1, 0));
+    TEST_DO(stats3.assertChanges(1, 0));
     f.removeListeners("testdoc", {"test1"});
-    TEST_DO(stats1.assertCounts(1, 1, 0, 1));
-    TEST_DO(stats2.assertCounts(1, 1, 1, 1));
-    TEST_DO(stats3.assertCounts(1, 1, 0, 1));
+    TEST_DO(stats1.assertListeners(1, 1, 0));
+    TEST_DO(stats2.assertListeners(1, 1, 1));
+    TEST_DO(stats3.assertListeners(1, 1, 0));
     f.removeListeners("testdoc", {});
-    TEST_DO(stats1.assertCounts(1, 1, 1, 1));
-    TEST_DO(stats2.assertCounts(1, 1, 1, 1));
-    TEST_DO(stats3.assertCounts(1, 1, 0, 1));
+    TEST_DO(stats1.assertListeners(1, 1, 1));
+    TEST_DO(stats2.assertListeners(1, 1, 1));
+    TEST_DO(stats3.assertListeners(1, 1, 0));
     f.removeListeners("testdoc2", {"test3"});
-    TEST_DO(stats1.assertCounts(1, 1, 1, 1));
-    TEST_DO(stats2.assertCounts(1, 1, 1, 1));
-    TEST_DO(stats3.assertCounts(1, 1, 0, 1));
+    TEST_DO(stats1.assertListeners(1, 1, 1));
+    TEST_DO(stats2.assertListeners(1, 1, 1));
+    TEST_DO(stats3.assertListeners(1, 1, 0));
     f.removeListeners("testdoc2", {"foo"});
-    TEST_DO(stats1.assertCounts(1, 1, 1, 1));
-    TEST_DO(stats2.assertCounts(1, 1, 1, 1));
-    TEST_DO(stats3.assertCounts(1, 1, 1, 1));
+    TEST_DO(stats1.assertListeners(1, 1, 1));
+    TEST_DO(stats2.assertListeners(1, 1, 1));
+    TEST_DO(stats3.assertListeners(1, 1, 1));
 }
 
 TEST_F("Test that we keep old listener when registering duplicate", Fixture)
 {
     auto &stats = f.addStats();
     auto listener = std::make_unique<MyListener>(stats, "test1", "testdoc");
-    TEST_DO(stats.assertCounts(1, 0, 0, 0));
+    TEST_DO(stats.assertListeners(1, 0, 0));
     f.addListener(std::move(listener));
-    TEST_DO(stats.assertCounts(1, 1, 0, 0));
+    TEST_DO(stats.assertListeners(1, 1, 0));
     listener = std::make_unique<MyListener>(stats, "test1", "testdoc");
-    TEST_DO(stats.assertCounts(2, 1, 0, 0));
+    TEST_DO(stats.assertListeners(2, 1, 0));
     f.addListener(std::move(listener));
-    TEST_DO(stats.assertCounts(2, 1, 1, 0));
+    TEST_DO(stats.assertListeners(2, 1, 1));
 }
 
 TEST_F("Test that put is ignored if we have a pending remove", Fixture)
 {
     auto &stats = f.addStats();
     auto listener = std::make_unique<MyListener>(stats, "test", "testdoc");
-    TEST_DO(stats.assertCounts(1, 0, 0, 0));
+    TEST_DO(stats.assertListeners(1, 0, 0));
```

Comment:
I don't think it is necessary to test listener counts in this test, already covered by other tests. Please consider removing them.